pkg/registrymirror/registrymirrorsstore_test.go

```diff
@@ -0,0 +1,339 @@
+			},
+		},
+	}
+
+	idmsA = &configV1.ImageDigestMirrorSet{
+		ObjectMeta: v1.ObjectMeta{Name: "idmsA", UID: "UIDidmsA"},
+		Spec: configV1.ImageDigestMirrorSetSpec{
+			ImageDigestMirrors: []configV1.ImageDigestMirrors{
+				{Source: "idms.registry.com", Mirrors: []configV1.ImageMirror{"idms.mirror1.com", "idms.mirror2.com"}},
+			},
+		},
+	}
+
+	itmsA = &configV1.ImageTagMirrorSet{
+		ObjectMeta: v1.ObjectMeta{Name: "itmsA", UID: "UIDitmsA"},
+		Spec: configV1.ImageTagMirrorSetSpec{
+			ImageTagMirrors: []configV1.ImageTagMirrors{
+				{Source: "itms.registry.com", Mirrors: []configV1.ImageMirror{"itms.mirror1.com", "itms.mirror2.com"}},
+			},
+		},
+	}
+)
+
+func fileContains(t *testing.T, path, text string) bool {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		t.Error(err)
+		return false
+	}
+
+	return strings.Contains(string(b), text)
+}
+
+func TestUpsertDelete(t *testing.T) {
+	path := filepath.Join(t.TempDir(), "registries.conf")
+
+	s := NewFileStore(WithConfigPath(path), WithDelay(0))
+	assert.Len(t, s.icspRules, 0)
+	assert.NoFileExists(t, path)
+
+	t.Run("ICSP", func(t *testing.T) {
+		source := icspA.Spec.RepositoryDigestMirrors[0].Source
+		err := s.UpsertImageContentSourcePolicy(icspA)
+		assert.NoError(t, err)
+		assert.Len(t, s.icspRules, 1)
+		assert.True(t, fileContains(t, path, source), "config missing registry")
+
+		err = s.DeleteImageContentSourcePolicy(icspA.UID)
+		assert.NoError(t, err)
+		assert.Len(t, s.icspRules, 0)
+		assert.False(t, fileContains(t, path, source), "config has registry but shouldn't")
+	})
+
+	t.Run("IDMS", func(t *testing.T) {
+		source := idmsA.Spec.ImageDigestMirrors[0].Source
+		err := s.UpsertImageDigestMirrorSet(idmsA)
+		assert.NoError(t, err)
+		assert.Len(t, s.idmsRules, 1)
+		assert.True(t, fileContains(t, path, source), "config missing registry")
+
+		err = s.DeleteImageDigestMirrorSet(idmsA.UID)
+		assert.NoError(t, err)
+		assert.Len(t, s.idmsRules, 0)
+		assert.False(t, fileContains(t, path, source), "config has registry but shouldn't")
+	})
+
+	t.Run("ITMS", func(t *testing.T) {
+		source := itmsA.Spec.ImageTagMirrors[0].Source
+		err := s.UpsertImageTagMirrorSet(itmsA)
+		assert.NoError(t, err)
+		assert.Len(t, s.itmsRules, 1)
+		assert.True(t, fileContains(t, path, source), "config missing registry")
+
+		err = s.DeleteImageTagMirrorSet(itmsA.UID)
+		assert.NoError(t, err)
+		assert.Len(t, s.itmsRules, 0)
+		assert.False(t, fileContains(t, path, source), "config has registry but shouldn't")
+	})
+}
+
+func TestDelayedUpdate(t *testing.T) {
+	path := filepath.Join(t.TempDir(), "registries.conf")
+
+	s := NewFileStore(WithConfigPath(path), WithDelay(200*time.Millisecond))
+	assert.Len(t, s.icspRules, 0)
+	assert.NoFileExists(t, path)
+
+	err := s.UpsertImageContentSourcePolicy(icspA)
+	assert.NoError(t, err)
+	assert.Len(t, s.icspRules, 1)
+	assert.NoFileExists(t, path)
+
+	time.Sleep(300 * time.Millisecond)
```

Comment:
nit: instead of the `time.Sleep` you could also use [assert.Eventually](https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually).

Reply:
Neat, didn't realize that API existed.  

For this particular test we know the 'delay' before condition should be true so the periodic check seems somewhat unnecessary (suppose the disk could be slow however).

Tried it out and IMO the test is less readable:

```go
waitFor := 1 * time.Second
checkEvery := 250 * time.Millisecond
conditionFn := func() bool { return fileContains(t, path, icspA.Spec.RepositoryDigestMirrors[0].Source) }
assert.Eventually(t, conditionFn, waitFor, checkEvery, "config missing registry")
```

vs.

```go
time.Sleep(300 * time.Millisecond)
assert.True(t, fileContains(t, path, icspA.Spec.RepositoryDigestMirrors[0].Source), "config missing registry")
```

Still, will make the update. Thanks.